src/oc/auth/email.clj

```diff
@@ -1,40 +1,222 @@
 (ns oc.auth.email
   (:require [clojure.string :as s]
+            [clojure.walk :refer (keywordize-keys)]
+            [if-let.core :refer (if-let*)]
+            [schema.core :as schema]
             [buddy.hashers :as hashers]
-            [oc.auth.config :as config]
+            [oc.lib.hateoas :as hateoas]
             [oc.auth.user :as user]))
 
-;(def ^:private prefix "email:")
+(def prefix "email-")
 
-(def auth-settings {:auth-url (s/join "/" [config/auth-server-url "email-auth"])
-                    :refresh-url (s/join "/" [config/auth-server-url "email" "refresh-token"])})
+(def ^:private crypto-algo "bcrypt+sha512$")
+
+;; ----- Schema -----
+
+; active - verified email
+; pending - awaiting invite response
+; unverified - awaiting email verification
+(def statuses #{"pending" "unverified" "active"})
+
+(def EmailUser 
+  (merge user/User {
+   :email schema/Str
+   :password-hash schema/Str
+   :status (schema/pred #(statuses %))
+   :auth-source (schema/pred #(= "email" %))}))
+
+(def updateable-props #{:email :password :name :first-name :last-name :real-name :avatar})
+(def public-props [:email :avatar :name :first-name :last-name :real-name])
+(def jwt-props [:user-id :org-id :name :first-name :last-name :avatar :email :auth-source])
```

Comment:
@belucid i think we should add :real-name here or the jwt will not have this field. it's important since it's used in some places of the UI (with fallback to :first-name and :last-name and in the subject of send email (the fallback here is to only ["You have been invited"](https://github.com/open-company/open-company-email/blob/29fc895dbe9edeccfaba6931721989613ef750d0/src/oc/email/mailer.clj#L72))